src/Vehicle/Vehicle.cc

```diff
@@ -3984,6 +3989,13 @@ void Vehicle::_updateHeadingToNextWP()
     }
 }
 
+void Vehicle::_updateWpIndexNumber()
+{
+    const int _currentIndex = _missionManager->currentIndex();
+
+    _wpIndexNumberFact.setRawValue(_currentIndex + 1);
```

Comment:
The "+1" must be based on !firmwarePlugin()->sendHomePositionToVehicle(). If this is false then you +1 the index. This accounts for a different in how PX4 and ArduPilot handle mission items. ArduPilot always has a fake home position item in index 0, PX4 does not. QGC always has a home position item in index 0. So item indices coming directly from mavlink must be massaged appropriately.